### ISSUE TITLE
package.json engines section parsed not properly

### ISSUE THREAD

**Issue by pumano** [yarnpkg/yarn#4998]
**Do you want to request a *feature* or report a *bug*?**
*bug*

**What is the current behavior?**
When some dependency contain engines section with node >=0.4.x yarn parsed it not properly.

Check https://github.com/shinuza/decimal-js/blob/master/package.json#L18 for that syntax

When I try to `yarn` with that dependency I got error message:
![image](https://user-images.githubusercontent.com/17675259/33271150-79ad9b16-d397-11e7-81d9-387c589d9006.png)

**If the current behavior is a bug, please provide the steps to reproduce.**
1) create any folder with package.json:

```
{
  "name": "test",
  "version": "1.0.0",
  "description": "description",
  "main": "index.js",
  "author": "Test",
  "license": "MIT",
  "private": true,
  "devDependencies": {
    "decimal": "0.0.2"
  }
}
```
2) run ```yarn```

**What is the expected behavior?**
Package should be installed properly

**Please mention your node.js, yarn and operating system version.**
node 8.8.1
npm 5.5.1
yarn 1.3.2
Windows 10 x64 / Linux x64

**Comment by rally25rs** [comment]
Confirmed on Yarn v1.3.2.

NPM installs decimal@0.0.2 without warning. Interesting this check passes semver too, which is what I would have expected Yarn to be using internally:

```
> const s = require('semver')
undefined
> s.satisfies('8.8.1', '>=0.4.x')
true
```

**Comment by rally25rs** [comment]
The published `decimal@0.0.2` package does not specify `>=` in its engines.

If you pull the data from the registry: http://registry.npmjs.org/decimal and look at the `engines` entry, it contains `"engines":{"node":"0.4.x"}`

If you install the package with `yarn add decimal@0.0.2 --ignore-engines` then check it's package.json:

```
$ cat node_modules/decimal/package.json
{
  "name": "decimal",
  "description": "Simple decimal arithmetic for the browser and node.js!",
  "author": "Samori Gorse <samorigorse@gmail.com>",
  "version": "0.0.2",
  "keywords": [
    "arithmetic",
    "decimal",
    "float",
    "calculation"
  ],
  "main": "./lib/decimal",
  "repository": {
    "type": "git",
    "url": "git://github.com/shinuza/decimal-js.git"
  },
  "engines": {
    "node": "0.4.x"
  }
}
```

So both the NPM registry and the actual package.json specify `0.4.x` (without the `>=`).

The actual Github repo for this library _does_ contain the `>=0.4.x` in package.json, but I suspect the package author did not publish that change, since it was made on 2015-11-24, and v0.0.2 appears to have been published on 2011-12-25. My recommendation would be to request the author to republish the package with a new version number.
